dhall/src/Dhall/TypeCheck.hs

```diff
@@ -3374,7 +3374,19 @@ prettyTypeMessage (MustMapARecord _expr0 _expr1) = ErrorMessages {..}
         \    └─────────────────────────────────────────────────────────────────────┘     \n\
         \                                                                                \n\
         \                                                                                \n\
-        \... but the argument to ❰toMap❱ must be a record and not some other type.       \n"
+        \... but the argument to ❰toMap❱ must be a record and not some other type.       \n\
+        \                                                                                \n\
+        \Some common reasons why you might get this error:                               \n\
+        \                                                                                \n\
+        \● You accidentally provide an empty record type instead of an empty record when \n\
+        \  using ❰toMap❱:                                                                \n\
+        \                                                                                \n\
+        \                                                                                \n\
+        \    ┌──────────┐                                                                \n\
+        \    │ toMap {} │                                                                \n\
```

Comment:
Can you add a type annotation? Otherwise it will still not typecheck after replacing `{}` with `{=}`.

Reply:
@sjakobi sure thing, good point!